src/components/cylc/tree/index.js

```diff
@@ -37,18 +37,49 @@ function createWorkflowNode (workflow) {
   }
 }
 
+/**
+ * Create a cycle point node ID. Uses the node property `id`.
+ *
+ * It will split the node ID using the separator. And if the number
+ * of tokens is at least three, it will return the three first tokens
+ * joined by the separator too.
+ *
+ * Otherwise returns the node ID.
+ *
+ * For example, given a node with the following ID's:
+ *
+ * - 'a|b|c|d'    results in a cycle point node ID 'a|b|c'
+ * - 'a|b|c|d|e'  results in a cycle point node ID 'a|b|c'
+ * - 'a|b|c'      results in a cycle point node ID 'a|b|c'
+ * - 'a|b'        results in a cycle point node ID 'a|b'
+ * - ''           results in a cycle point node ID ''
+ *
+ * @param node {{
+ *   id: string
+ * }} a tree node
+ * @return {string}
+ */
+function getCyclePointId (node) {
+  const tokens = node.id.split('|')
+  if (tokens.length >= 3) {
+    return tokens.splice(0, 3).join('|')
+  }
+  return node.id
```

Comment:
Normally I would `throw new Error`, but errors are now displayed to the user... need to think if we could have a token with less than 3 parts (e.g. workflow, or a token with an invalid ID) in the deltas... :thinking: 

If I can't think of a good case, I will change it to `throw new Error`, as that way at least we will be notified about the error and be able to fix it.